python/dgl/distributed/graph_partition_book.py

```diff
@@ -793,27 +828,19 @@ def __init__(self, part_id, num_parts, node_map, edge_map, ntypes, etypes):
             ntype is not None for ntype in self._ntypes
         ), "The node types have invalid IDs."
         for etype, etype_id in etypes.items():
-            if isinstance(etype, tuple):
-                assert (
-                    len(etype) == 3
-                ), "Canonical etype should be in format of (str, str, str)."
-                c_etype = etype
-                etype = etype[1]
-                self._etypes[etype_id] = etype
-                self._canonical_etypes[etype_id] = c_etype
-                if etype in self._etype2canonical:
-                    # If one etype maps to multiple canonical etypes, empty
-                    # tuple is used to indicate such ambiguity casued by etype.
-                    # See more details in self._to_canonical_etype().
-                    self._etype2canonical[etype] = tuple()
-                else:
-                    self._etype2canonical[etype] = c_etype
+            assert isinstance(etype, tuple) and len(etype) == 3, \
+                "Only canonical etype in tuple is supproted: (str, str, str)."
+            c_etype = etype
```

Comment:
Could you rename the loop variable at L830 to c_etype? Then you can delete L833 and change L834 to `etype = c_etype[1]`.